src/audio/kpb.c

```diff
@@ -171,45 +179,52 @@ static int kpb_prepare(struct comp_dev *dev)
 
 	cd->no_of_clients = 0;
 
-	/* allocate history_buffer/s */
-#if KPB_NO_OF_HISTORY_BUFFERS == 2
+	/* allocate history buffer/s */
+	while (hb_size > 0 && i < ARRAY_SIZE(hb_mcp)) {
```

Comment:
This has potential to panic since we are allocating a buffer in a loop. Can we validate hb_size and hb_mcp before entering the loop.

Reply:
@lgirdwood, I don't get it, we have initialized both of these variables right before the start of while loop, so why would you validate them?

Reply:
@mrajwa this part is quite confusing. Are you trying to allocate a list of buffers here which add up to the size KPB_MAX_BUFFER_SIZE?

Reply:
Correct.